contracts/Tokens/Prime/Prime.sol

```diff
@@ -0,0 +1,908 @@
+     * @return cappedBorrow the capped borrow of user
+     */
+    function _capitalForScore(
+        uint256 xvs,
+        uint256 borrow,
+        uint256 supply,
+        address market
+    ) internal view returns (uint256, uint256, uint256) {
+        address xvsToken = IXVSVault(xvsVault).xvsAddress();
+
+        uint256 xvsPrice = oracle.getPrice(xvsToken);
+        uint256 borrowCapUSD = (xvsPrice * ((xvs * markets[market].borrowMultiplier) / EXP_SCALE)) / EXP_SCALE;
+        uint256 supplyCapUSD = (xvsPrice * ((xvs * markets[market].supplyMultiplier) / EXP_SCALE)) / EXP_SCALE;
+
+        uint256 tokenPrice = oracle.getUnderlyingPrice(market);
+        uint256 supplyUSD = (tokenPrice * supply) / EXP_SCALE;
+        uint256 borrowUSD = (tokenPrice * borrow) / EXP_SCALE;
+
+        if (supplyUSD >= supplyCapUSD) {
+            supply = supplyUSD > 0 ? (supply * supplyCapUSD) / supplyUSD : 0;
+        }
+
+        if (borrowUSD >= borrowCapUSD) {
+            borrow = borrowUSD > 0 ? (borrow * borrowCapUSD) / borrowUSD : 0;
+        }
+
+        return ((supply + borrow), supply, borrow);
+    }
+
+    /**
+     * @notice Used to mint a new prime token
+     * @param isIrrevocable is the tokens being issued is irrevocable
+     * @param user token owner
+     */
+    function _mint(bool isIrrevocable, address user) internal {
+        if (tokens[user].exists) revert IneligibleToClaim();
+
+        tokens[user].exists = true;
+        tokens[user].isIrrevocable = isIrrevocable;
+
+        if (isIrrevocable) {
+            totalIrrevocable++;
+        } else {
+            totalRevocable++;
+        }
+
+        if (totalIrrevocable > irrevocableLimit || totalRevocable > revocableLimit) revert InvalidLimit();
+
+        emit Mint(user, isIrrevocable);
+    }
+
+    /**
+     * @notice Used to burn a new prime token
+     * @param user owner whose prime token to burn
+     */
+    function _burn(address user) internal {
+        if (!tokens[user].exists) revert UserHasNoPrimeToken();
+
+        address[] storage _allMarkets = allMarkets;
+
+        for (uint i = 0; i < _allMarkets.length;) {
+            _executeBoost(user, _allMarkets[i]);
+
+            markets[_allMarkets[i]].sumOfMembersScore =
+                markets[_allMarkets[i]].sumOfMembersScore -
+                interests[_allMarkets[i]][user].score;
+            interests[_allMarkets[i]][user].score = 0;
+            interests[_allMarkets[i]][user].rewardIndex = 0;
+
+            unchecked {
+                i++;
+            }
+        }
+
+        if (tokens[user].isIrrevocable) {
+            totalIrrevocable--;
+        } else {
+            totalRevocable--;
+        }
+
+        tokens[user].exists = false;
+        tokens[user].isIrrevocable = false;
+
+        _updateRoundAfterTokenBurned(user);
+
+        emit Burn(user);
+    }
+
+    /**
+     * @notice Used to get if the XVS balance is eligible for prime token
+     * @param amount amount of XVS
```

Comment:
```suggestion
     * @param amount amount of XVS
     * @return isEligible true if the staked XVS amount is enough to consider the associated user eligible for a Prime token, false otherwise
```